src/components/App/App.js

```diff
@@ -0,0 +1,353 @@
+  getFavoriteArtists,
+  getFavoriteAlbums,
+  getFavoriteTracks,
+  addFavortieTrack,
+  addFavortieArtist,
+  addFavortieAlbum,
+  deleteFavoriteAlbum,
+  deleteFavoriteTrack,
+  deleteFavoriteArtist,
+} from "../../utils/api";
+import arrow from "../../images/uparrow.png";
+
+function App() {
+  const history = useHistory();
+  const [activeModal, setActiveModal] = useState("");
+  const [authToken, setAuthToken] = useState("");
+  const [topArtistData, setTopArtistData] = useState([]);
+  const [newReleaseData, setNewReleaseData] = useState([]);
+  const [recommendations, setRecommendations] = useState([]);
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [favoriteArtists, setFavoriteArtists] = useState([]);
+  const [favoriteTracks, setFavoriteTracks] = useState([]);
+  const [favoriteAlbums, setFavoriteAlbums] = useState([]);
+  const [isVisible, setIsVisible] = useState(false);
+
+  const handleCloseModal = () => {
+    setActiveModal("");
+  };
+
+  const handleEditModal = () => {
+    setActiveModal("edit");
+  };
+
+  const handleLoginModal = () => {
+    setActiveModal("login");
+  };
+
+  const handleRegisterModal = () => {
+    setActiveModal("register");
+  };
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth", 
+    });
+  };
+
+  useEffect(() => {
+    getToken()
+      .then((data) => {
+        setAuthToken(data.access_token);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, []);
+
+  useEffect(() => {
+    // if not authToken, then don't make the request
+    if (authToken === "") {
+      return;
+    } else {
+      getTopArtistData(authToken)
+        .then((data) => {
+          setTopArtistData(data.artists.items);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+
+      getNewReleaseData(authToken)
+        .then((data) => {
+          setNewReleaseData(data.albums.items);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+
+      getRecommendations(authToken)
+        .then((res) => {
+          setRecommendations(res?.albums?.items);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  }, [authToken]);
+
+  useEffect(() => {
+    const token = localStorage.getItem("jwt");
+
+    if (token) {
+      checkToken(token)
+        .then((data) => {
+          setCurrentUser(data);
+          setLoggedIn(true);
+        })
+        .catch(() => console.error);
+    } else {
+      localStorage.removeItem("jwt");
+      setLoggedIn(false);
+    }
+  }, [loggedIn, history]);
+
+  useEffect(() => {
+    if (loggedIn) {
+      getFavoriteArtists()
+        .then((res) => {
+          setFavoriteArtists(res);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+      getFavoriteTracks().then((res) => {
+        setFavoriteTracks(res);
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**